cmake/modules/AddSwift.cmake

```diff
@@ -140,7 +140,11 @@ function(_add_variant_c_compile_link_flags)
     # lld can handle targeting the android build.  However, if lld is not
     # enabled, then fallback to the linker included in the android NDK.
     if(NOT SWIFT_ENABLE_LLD_LINKER)
-      list(APPEND result "-B" "${SWIFT_SDK_ANDROID_ARCH_${CFLAGS_ARCH}_NDK_PREBUILT_PATH}/${SWIFT_SDK_ANDROID_ARCH_${CFLAGS_ARCH}_NDK_TRIPLE}/bin")
+      if("${SWIFT_HOST_VARIANT}" STREQUAL "android")
+        list(APPEND result "-B" "${ANDROID_NATIVE_SYSROOT}/usr/bin")
```

Comment:
We can initialize `ANDROID_NATIVE_SYSROOT` to the NDK path as we used to if you like, but, I think that computing it is better as it allows for cross-compiling to different target architectures.

Reply:
I'm not sure what you mean, but I'm guessing this is because you think SWIFT_ANDROID_NDK_PATH is set and being used by this pull. I pointed out in a separate comment above that I'm not using it.

Reply:
Right, I am suggesting that we figure out how to get that that setup properly.  This will enable us to share the codepath for the cross-compile and native compile paths.

Reply:
Even if I got rid of ANDROID_NATIVE_SYSROOT and set SWIFT_ANDROID_NDK_PATH to the same Termux path on Android, instead of leaving the latter uninitialized like I am now, the paths used for headers and libraries are different in the Termux app than the NDK, so the different CMake config here would still be needed.